### PR TITLE
Ft/ammo

### PR DESCRIPTION
**Pull request by LouisAndrew**
- [x]  Display ammo in a Canvas UI.
- [x]  Ammo should be decreased if a projectile is instantiated
- [x]  Disable shooting if ammo ≤ 0
- [x]  Display amount of ammo in real-time
- [x] Create mock list of `weaponData` -> to be used on `PlayerSave`. 
- [x] Assign every weapon with an ID -> useful for assigning names and keeping track of its ammo.

### CODE REVIEW

File: Assets/Scripts/Shooting/ShootingController.cs [hunk 1]
```diff
@@ -17,12 +23,15 @@ void Start()
...
+
+        _ammo = _maxAmmo;
+        _weaponHolder = GetComponentInParent<WeaponHolder>();
     }
 
     // Update is called once per frame
     void Update()
     {
-        if (Input.GetButtonDown("Fire1"))
+        if (Input.GetButtonDown("Fire1") && _ammo != 0)
```
Comment (hunsri) [inline Assets/Scripts/Shooting/ShootingController.cs]: But what happens in case of _ammo < 0

**Review by hunsri — CHANGES_REQUESTED** [review]
After weapon change the ammo is refilled. I would change that.
Also the current check for ammo could result in "endless ammo" in case the amount drops to -1 for some reason.

File: Assets/Scripts/Shooting/ShootingController.cs (hunk 1 above)
Comment (LouisAndrew) [reply Assets/Scripts/Shooting/ShootingController.cs]: yeah will change those

**Comment by LouisAndrew** [comment]
> After weapon change the ammo is refilled. I would change that.
> Also the current check for ammo could result in "endless ammo" in case the amount drops to -1 for some reason.

Yeah, ammo is refilled for now. The part where the ammo count would not be refilled after weapon change is going to be implemented next in "Save and Load game".

**Comment by LouisAndrew** [comment]
Also it's part of `PlayerData`, which is not yet implemented and that needs this function to first be merged with master.

**Review by hunsri — APPROVED** [review]
I think that's all :+1:

**Pull request merged**